aiohttp_retry/client.py

```diff
@@ -108,38 +113,31 @@ async def _do_request(self) -> ClientResponse:
                     **(params.kwargs or {}),
                 )
 
-                if self._is_status_code_ok(response.status) or current_attempt == self._retry_options.attempts:
+                retry_message = f"Retrying after response code: {response.status}"
+                skip_retry = (current_attempt == self._retry_options.attempts
```

Comment:
I don't like when ```await``` is a part of complex bool statement. It just feels ugly
I would suggest you to do a next thing:
```python
skip_retry = await self._is_skip_retry(
    current_attempt=current_attempt,
    retry_options=retry_options,
    response=response,
)
```

However I'm totally agree on your idea that this logic should be put in a separate function